lib/package-managers/npm.js

```diff
@@ -107,25 +117,57 @@ module.exports = {
             npm.commands.listAsync(args || [], true); // silent:true
     },
 
-    latest: cint.partialAt(view, 1, 'dist-tags.latest'),
+    latest: function (packageName, currentVersion, includePrerelease) {
+        return view(packageName, 'dist-tags.latest')
+            .then(function (version) {
+                if (
+                    versionUtil.getPrecision(version) !== 'release' ||
+                    includePrerelease
+                ) {
+                    return version;
+                } else {
+                    return view(packageName, 'versions')
+                        .then(filterOutPrereleaseVersions)
+                        .then(_.last);
+                }
+            });
+    },
 
-    newest: function (packageName) {
+    newest: function (packageName, currentVersion, includePrerelease) {
         return view(packageName, 'time')
             .then(_.keys)
-            .then(_.last);
+            .then(function (versions) {
+                if (!includePrerelease) {
+                    versions = filterOutPrereleaseVersions(versions);
```

Comment:
This is bad practice to overwrite a local variable with a different value. Could you combine these into a single return statement?

```
return _last(includePrelease ? versions : filterOutPrereleaseVersions(versions));
```